### ISSUE TITLE
Renderdoc Crashes when trying to open Vulkan capture

### ISSUE THREAD

**Issue by YVin3D** [baldurk/renderdoc#790]
So I have [this repo](https://github.com/YoussefV/Project6-Vulkan-Grass-Rendering) That is in Vulkan. This is just the base code for an assignment. (Use Cmake and VS to run)

When I trigger a capture for the release mode of this project, and then open that trigger, RenderDoc crashes.

Here is the log file:

[RenderDoc_2017.11.01_20.18.24.log](https://github.com/baldurk/renderdoc/files/1436085/RenderDoc_2017.11.01_20.18.24.log)

and here is the capture file:

[vulkan_grass_rendering_2017.11.01_20.18.38_frame231.zip](https://github.com/baldurk/renderdoc/files/1436087/vulkan_grass_rendering_2017.11.01_20.18.38_frame231.zip)

If I try to debug the renderdoc crash in VS, I get "heap corruption". But the weird thing is that the same code works on my friends' computers, some with the same GPU as well. On a whole different range of PC's, even one with a virtual machine set up on his mac. So I have no idea why it's just me.

My Specs: Quad Core i7-7700HQ, GeForce GTX 1050 2G Graphics, 8GB DRAM

Any help would be appreciated. Thanks!

**Comment by baldurk** [comment]
I believe this is related to the mention in the log of two physical devices being available on capture (Intel & NV GPUs) and only one on replay. It looks like there was a bug introduced at some point, for some cases trying to replay a capture made with 2 GPUs on a single GPU. I've committed a fix so the next nightly build should load the capture fine.

As an aside, the disappearing intel GPU is due to the buggy/awful NV optimus layer. I've also added a commit to set the environment variable to force-disable it on replay, so hopefully this won't be an issue in future in case someone writes code that uses both GPUs or chooses the Intel GPU.
